x-pack/plugins/endpoint/public/applications/endpoint/view/alerts/details/overview/take_action_dropdown.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo, useMemo, useState } from 'react';
+import { EuiPopover, EuiFormRow, EuiButton, EuiButtonEmpty } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+export const TakeActionDropdown = memo(() => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+  const TakeActionButton = useMemo(() => {
```

Comment:
Creating a new component in the render method of another component is a bit nonstandard. The component itself will be redefined when `isDropdownOpen` is changed. For consistency with the recommended react patterns, consider defining this component once.

Reply:
Here's a commit with the changes I'd recommend: https://github.com/oatkiller/kibana/commit/98bca4a1e4b30df40a8fca53e4edde359b48030a